### PR TITLE
[JBPM-9434] REST endpoint authetication fails if Authorization header…

### PR DESCRIPTION
**Pull request by abhijithumbe**
… has == in encoded key in REST WIH

**Thank you for submitting this pull request**

**JIRA**: _(https://issues.redhat.com/browse/JBPM-9434)_

### CODE REVIEW

File: jbpm-workitems/jbpm-workitems-rest/src/main/java/org/jbpm/process/workitem/rest/RESTWorkItemHandler.java [hunk 1]
```diff
@@ -905,7 +905,7 @@ protected void addAcceptHeader(HttpRequestBase theMethod,
     protected void addCustomHeaders(String headers,
             BiConsumer<String, String> consumer) {
         for(String h : headers.split(";")) {
-             String[] headerParts = h.split("=");
+             String[] headerParts = h.split(":");
```
Comment (gmunozfe) [inline jbpm-workitems/jbpm-workitems-rest/src/main/java/org/jbpm/process/workitem/rest/RESTWorkItemHandler.java:908]: We can keep previous separator, but forcing the length of the resulting array, just to split with the first separator:
```suggestion
             String[] headerParts = h.split("=", 2);
```

File: jbpm-workitems/jbpm-workitems-rest/src/test/java/org/jbpm/process/workitem/rest/RestWorkItemHandlerTest.java [hunk 2]
```diff
@@ -1069,7 +1069,29 @@ public void testHeadersSingleValue() {
...
+        workItem.setParameter("Method",
+                              "GET");
+        workItem.setParameter(PARAM_HEADERS, headers);
+        
+        WorkItemManager manager = new TestWorkItemManager();
+        handler.executeWorkItem(workItem,
+                                manager);
+        Map<String, Object> results = ((TestWorkItemManager) manager).getResults(workItem.getId());
+        String result = (String) results.get(PARAM_RESULT);
+        assertEquals(result, headerValue);
```
Comment (gmunozfe) [inline jbpm-workitems/jbpm-workitems-rest/src/test/java/org/jbpm/process/workitem/rest/RestWorkItemHandlerTest.java:1086]: It should be the other way around (expected, current):
```suggestion
        assertEquals(headerValue, result);
```

**Review by gmunozfe — CHANGES_REQUESTED** [review]
hi @abhijithumbe, thanks for pointing this out.
If the issue is because the separator appears more than one time, and only the first one is relevant, I think we could use the `split("=", 2) ` to get just 2 elements as expected. Therefore, we don't break backward compatibility, by using the same separator, wdyt?
You can keep the test that you have added, just one comment there

File: jbpm-workitems/jbpm-workitems-rest/src/main/java/org/jbpm/process/workitem/rest/RESTWorkItemHandler.java (hunk 1 above)
Comment (abhijithumbe) [reply jbpm-workitems/jbpm-workitems-rest/src/main/java/org/jbpm/process/workitem/rest/RESTWorkItemHandler.java]: @gmunozfe Thank you for pointing it out. I should have thought about backward compatibility.

File: jbpm-workitems/jbpm-workitems-rest/src/test/java/org/jbpm/process/workitem/rest/RestWorkItemHandlerTest.java (hunk 2 above)
Comment (abhijithumbe) [reply jbpm-workitems/jbpm-workitems-rest/src/test/java/org/jbpm/process/workitem/rest/RestWorkItemHandlerTest.java]: @gmunozfe Made suggested changes in this test case and in other test cases as well.

**Review by gmunozfe — APPROVED** [review]
Looks good to me, well done @abhijithumbe !

**Comment by abhijithumbe** [comment]
hi @kiegroup/gatekeepers, could you merge?

**Comment by gmunozfe** [comment]
@kiegroup/gatekeepers could you merge please?

**Comment by mareknovotny** [comment]
jenkins retest this please

**Comment by abhijithumbe** [comment]
@mareknovotny can this PR be merged or need any modification ?

**Pull request merged**